src/realm/alloc.hpp

```diff
@@ -495,25 +539,40 @@ inline Allocator::~Allocator() noexcept
 {
 }
 
-inline char* Allocator::translate(ref_type ref) const noexcept
+// performance critical part of the translation process. Less critical code is in translate_less_critical.
+inline char* Allocator::translate_critical(RefTranslation* ref_translation_ptr, ref_type ref) const noexcept
 {
-    if (auto ref_translation_ptr = m_ref_translation_ptr.load(std::memory_order_acquire)) {
-        char* base_addr;
-        size_t idx = get_section_index(ref);
-        base_addr = ref_translation_ptr[idx].mapping_addr;
-        size_t offset = ref - get_section_base(idx);
-        auto addr = base_addr + offset;
+    size_t idx = get_section_index(ref);
+    RefTranslation& txl = ref_translation_ptr[idx];
+    size_t offset = ref - get_section_base(idx);
+    size_t mapping_limit = txl.lowest_possible_xover_offset.load(std::memory_order_relaxed);
```

Comment:
I thought you got rid of the mapping_limit names?

Reply:
whoops.